### ISSUE TITLE
Prebuilt package throws "TypeError: Cannot read property 'call' of undefined"

### ISSUE THREAD

**Issue by jamescryer** [Automattic/node-canvas#1310]
## Issue or Feature
I guess this is an issue with the pre-built fallback packages for certain environments. Also worth being aware that in my more complex application (not shown) I saw intermittent success of tests and functionality.

```
      TypeError: Cannot read property 'call' of undefined
          at setSource (node_modules/canvas/lib/image.js:102:14)
          at Image.set (node_modules/canvas/lib/image.js:68:7)
          at expect (nodejs-tests/canvas.test.js:19:39)
          at Object.test (nodejs-tests/canvas.test.js:19:53)
```

## Steps to Reproduce
These simple Jest tests fail on my machine.

```js
/* eslint-env jest*/
const fs = require("fs");
const util = require("util");
const canvas = require("canvas");
const readFile = util.promisify(fs.readFile);

describe("Canvas test", () => {
    test("loadImage", async () => {
        expect.assertions(1);
        const data = await readFile("./image.png");
        await expect(canvas.loadImage(data)).rejects.not.toBeTruthy();
    });
    test("Set Image.src", async () => {
        expect.assertions(1);
        const data = await readFile("./image.png");
        const image = new canvas.Image;
        await expect( () => { image.src = data;} ).not.toThrowError();
    });
});    
```

## Your Environment
* Version of node-canvas: 2.1.0 using prebuilt package (I haven't tried building locally)
* Environment: node v10.14.0 on Windows 10 Enterprise

**Comment by LinusU** [comment]
Could you post the entire stack trace? Thanks

**Comment by jamescryer** [comment]
@LinusU, updated, thanks!

**Comment by zbjornson** [comment]
(This is the same error as #1294 and #1250, both unresolved.)

I can finally repro this (thanks!), and it seems to be because of Jest 😞 . The following works with mocha:

```js
const fs = require("fs");
const util = require("util");
const canvas = require("canvas");
const readFile = util.promisify(fs.readFile);
const {expect} = require("chai");

describe("Canvas test", () => {
    it("Set Image.src", async () => {
        const data = await readFile("./image.png");
        const image = new canvas.Image;
        await expect( () => { image.src = data;} ).to.not.throw();
    });
});
```

I think this is because Jest breaks [node's guarantee](https://nodejs.org/dist/latest-v10.x/docs/api/modules.html#modules_caching) that `require()`'d modules are loaded exactly once and returned from cache on subsequent `require()` calls. With jest, `lib/Image.js` tries to store a reference to function that was deleted the first time the file is loaded.

(I've [complained about this](https://github.com/facebook/jest/issues/4413) to the Jest folks and they won't change the behavior. It's caused problems in most public repos I've worked on though.)

As much as I don't want to pander to Jest, we can move the `setSource`/`getSource` methods off of the prototype to avoid this, I think.

---

@chearon it's defined there in the native code, but deleted here in the JS:
https://github.com/Automattic/node-canvas/blob/65ff59a4e982a8c3582aef362d6c07c90ce57d7e/lib/image.js#L23

**Comment by LinusU** [comment]
Should be fixed in `2.2.0` 🚀
